src/book/02-system/01-foundations/07-director.mdx

```diff
@@ -0,0 +1,447 @@
+
+For example:
+
+```cpp
+on<Stop<Walk>>().then({ // a Stop Provider type for the Walk Task
+    // Log that the walk has stopped
+    log<NUClear::TRACE>("Walk Stopped");
+});
+```
+
+### Needs
+
+The Needs DSL word is used to ensure a Provider can take control of subtasks, otherwise it cannot run.
+
+```cpp
+on<Provide<Walk>, Needs<LegIK>>().then(...);
+```
+
+In this example, the Walk Provider will only run if it can take over the LegIK Task.
+
+If another, higher priority, Provider has already taken over the LegIK Task, then this Provider will not run. If the LegIK is not being used by any other Provider, or it has higher priority than the Provider controlling the LegIK, then this Provider will take over the LegIK Task.
+
+Without `Needs`, the Provider will still run and can emit subtasks, but the subtasks will not do anything if another Provider has access to them.
+
+### When
+
+A When condition is used to ensure that a Provider will only execute when the specified condition is met.
+These conditions are provided using a global enum message along with a statement that describes what condition you want.
+For example, if you wanted a kick engine to only run if the robot is statically stable or better then you would write:
+
+```cpp
+on<Provide<Kick>, When<Stability, std::greater_equal, Stability::STANDING>>().then(...);
+```
+
+These conditions are updated by using normal NUClear emits. For example if
+
+```cpp
+emit(std::make_unique<Stability>(Stability::STATIC));
+```
+
+was emitted in another reaction, then the `When` condition would be met and the Provider would run.
+
+### Causing
+
+A Causing statement is a promise that by running this Provider, a specified desired state will be eventually reached.
+This is used to allow smooth transitions between modules.
+For example, when the walk engine stops moving it would be in a `STANDING` Stability state.
+This means the walk engine could have a `Provide` reaction that causes this state:
+
+```cpp
+on<Provide<Walk>, Causing<Stability, Stability::STANDING>().then({
+    // Make the robot stop walking
+    set_orders = Eigen::Vector3d::Zero();
+    // Check if we have stopped moving
+    if (state == STOPPED) {
+        // Emit that we are standing
+        emit(std::make_unique<Stability>(Stability::STANDING));
+    }
+});
+```
+
+The implementation for the reaction should result in the desired state being eventually emitted. This would not be the primary Provider for the Walk, but would instead be actie when a higher priority Provider wants this state.
+
+For example, if the kick engine only runs `When<Stability, std::greater_equal, Stability::STANDING>` then it would not be able to run while the walk engine is running normally. However, if the kick Task has a higher priority than the walk Task, then it will push the walk engine into the Causing Provide reaction, which will then cause the Stability state to be STANDING so that the kick engine can run.
```

Comment:
```suggestion
For example, if the kick engine only runs `When<Stability, std::greater_equal, Stability::STANDING>` then it would not be able to run while the walk engine is running normally. However, if the kick Task has a higher priority than the walk Task, then it will push the walk engine into the Causing Provide reaction, which will then cause the Stability state to be `STANDING` so that the kick engine can run.
```